cryosparc/mrc.py

```diff
@@ -129,7 +129,7 @@ def write(file: Union[str, PurePath, IO[bytes]], data: "NDArray", psize: float):
 
     with bopen(file, "wb") as f:
         _write_header(f, data, psize)
-        n.require(data, requirements="C").ravel().tofile(f)
+        n.require(data, dtype=n.float32, requirements="C").ravel().tofile(f)
```

Comment:
I think the header will be incorrect if you do this, should do something like:
```py
data = n.require(data, dtype=n.float32, requirements="C")
_write_header(f, data, psize)
data.ravel.tofile(f)
```

Reply:
And one more thing, this cast should not happen if the data has type `n.float16`.